jbox2d-library/src/main/java/org/jbox2d/collision/broadphase/DefaultBroadPhaseBuffer.java

```diff
@@ -245,21 +245,19 @@ public final boolean treeCallback(int proxyId) {
 
     // Grow the pair buffer as needed.
     if (m_pairCount == m_pairCapacity) {
-      Pair[] oldBuffer = m_pairBuffer;
+      long[] oldBuffer = m_pairBuffer;
       m_pairCapacity *= 2;
-      m_pairBuffer = new Pair[m_pairCapacity];
+      m_pairBuffer = new long[m_pairCapacity];
       System.arraycopy(oldBuffer, 0, m_pairBuffer, 0, oldBuffer.length);
       for (int i = oldBuffer.length; i < m_pairCapacity; i++) {
-        m_pairBuffer[i] = new Pair();
+        m_pairBuffer[i] = 0;
       }
     }
 
     if (proxyId < m_queryProxyId) {
-      m_pairBuffer[m_pairCount].proxyIdA = proxyId;
-      m_pairBuffer[m_pairCount].proxyIdB = m_queryProxyId;
+      m_pairBuffer[m_pairCount] = ((long) proxyId << 32) + m_queryProxyId;
```

Comment:
Can you change this (and the line below) to use the "|" or operator instead of "+"? This helps me know that this is supposed to be a bitwise operation.

Reply:
Sure, done.